Mathlib/Analysis/InnerProductSpace/Basic.lean

```diff
@@ -360,6 +440,88 @@ theorem norm_inner_le_norm (x y : F) : ‖⟪x, y⟫‖ ≤ ‖x‖ * ‖y‖ :=
+      add_le' := fun x y => by
+        have h₁ : ‖⟪x, y⟫‖ ≤ ‖x‖ * ‖y‖ := norm_inner_le_norm _ _
+        have h₂ : re ⟪x, y⟫ ≤ ‖⟪x, y⟫‖ := re_le_norm _
+        have h₃ : re ⟪x, y⟫ ≤ ‖x‖ * ‖y‖ := h₂.trans h₁
+        have h₄ : re ⟪y, x⟫ ≤ ‖x‖ * ‖y‖ := by rwa [← inner_conj_symm, conj_re]
+        have : ‖x + y‖ * ‖x + y‖ ≤ (‖x‖ + ‖y‖) * (‖x‖ + ‖y‖) := by
+          simp only [← inner_self_eq_norm_mul_norm, inner_add_add_self, mul_add, mul_comm, map_add]
+          linarith
+        exact nonneg_le_nonneg_of_sq_le_sq (add_nonneg (sqrt_nonneg _) (sqrt_nonneg _)) this }
+
+attribute [local instance] toSeminormedAddCommGroup
+
+/-- Normed space (which is actually a seminorm) structure constructed from an
+`PreInnerProductSpace.Core` structure -/
+def toSeminormedSpace : NormedSpace 𝕜 F where
+  norm_smul_le r x := by
+    rw [norm_eq_sqrt_inner, inner_smul_left, inner_smul_right, ← mul_assoc]
+    rw [RCLike.conj_mul, ← ofReal_pow, re_ofReal_mul, sqrt_mul, ← ofReal_seminormSq_eq_inner_self,
+      ofReal_re]
+    · simp [sqrt_normSq_eq_norm, RCLike.sqrt_normSq_eq_norm]
+    · positivity
+
+end PreInnerProductSpace.Core
+
+section InnerProductSpace.Core
+
+variable [AddCommGroup F] [Module 𝕜 F] [cd : InnerProductSpace.Core 𝕜 F]
+
+local notation "⟪" x ", " y "⟫" => @inner 𝕜 F _ x y
+
+local notation "normSqK" => @RCLike.normSq 𝕜 _
+
+local notation "reK" => @RCLike.re 𝕜 _
+
+local notation "ext_iff" => @RCLike.ext_iff 𝕜 _
+
+local postfix:90 "†" => starRingEnd _
+
+/-- Inner product defined by the `InnerProductSpace.Core` structure. We can't reuse
+`InnerProductSpace.Core.toInner` because it takes `InnerProductSpace.Core` as an explicit
+argument. -/
+def toInner' : Inner 𝕜 F :=
+  cd.toInner
+
+attribute [local instance] toInner'
+
+/-- The norm squared function for `InnerProductSpace.Core` structure. -/
+def normSq (x : F) :=
+  reK ⟪x, x⟫
+#align inner_product_space.core.norm_sq InnerProductSpace.Core.normSq
+
+local notation "normSqF" => @normSq 𝕜 F _ _ _ _
+
+theorem inner_self_eq_zero {x : F} : ⟪x, x⟫ = 0 ↔ x = 0 :=
+  ⟨cd.definite _, by
+    rintro rfl
+    exact inner_zero_left _⟩
+#align inner_product_space.core.inner_self_eq_zero InnerProductSpace.Core.inner_self_eq_zero
+
+theorem normSq_eq_zero {x : F} : normSqF x = 0 ↔ x = 0 :=
+  Iff.trans
+    (by simp only [normSq, ext_iff, map_zero, inner_self_im, eq_self_iff_true, and_true_iff])
+    (@inner_self_eq_zero 𝕜 _ _ _ _ _ x)
+#align inner_product_space.core.norm_sq_eq_zero InnerProductSpace.Core.normSq_eq_zero
+
+theorem inner_self_ne_zero {x : F} : ⟪x, x⟫ ≠ 0 ↔ x ≠ 0 :=
+  inner_self_eq_zero.not
+#align inner_product_space.core.inner_self_ne_zero InnerProductSpace.Core.inner_self_ne_zero
+
+/-- Norm constructed from an `InnerProductSpace.Core` structure, defined to be the square root
+of the scalar product. -/
+def toNorm : Norm F where norm x := √(re ⟪x, x⟫)
+#align inner_product_space.core.to_has_norm InnerProductSpace.Core.toNorm
+
+attribute [local instance] toNorm
```

Comment:
Similarly, I think you don't need a new definition. You should just be able to make the `PreInnerProductSpace.Core.toNorm` a local instance.